src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -125,6 +156,27 @@ private static CSharpCompilation MetadataView(CSharpCompilation comp, params Met
             return CreateCompilation("", references: references);
         }
 
+        private static CSharpCompilation RetargetedView(CSharpCompilation comp, params MetadataReference[] references)
+        {
+            var someType = comp.GlobalNamespace.GetMembers().OfType<NamedTypeSymbol>().FirstOrDefault();
+            references = references.Append(comp.ToMetadataReference());
+            var result = CreateCompilation("", references: references, targetFramework: TargetFramework.NetStandardLatest);
+            result.VerifyDiagnostics();
+            var originalCorLib = comp.Assembly.CorLibrary;
+            var newCorLib = result.Assembly.CorLibrary;
+            Assert.NotEqual(originalCorLib, newCorLib);
+
+            if (someType is { })
+            {
+                // confirm that it is retargeted
+                var someRetargetedType = result.GlobalNamespace.GetMembers(someType.Name).OfType<NamedTypeSymbol>().FirstOrDefault();
```

Comment:
>result.GlobalNamespace [](start = 41, length = 22)

Instead of looking across the world, we could look in AssemblySymbol corresponding to the metadata reference for ```comp```. The type must be in that assembly.